hawkbit-repository/hawkbit-repository-jpa/src/test/java/org/eclipse/hawkbit/repository/jpa/RolloutManagementTest.java

```diff
@@ -1823,6 +1826,56 @@ public void createInSingleStartInMultiassigMode() {
+
+        assertThatExceptionOfType(InvalidDistributionSetException.class)
+                .as("Invalid distributionSet should throw an exception")
+                .isThrownBy(() -> testdataFactory.createRolloutByVariables("createRolloutWithInvalidDistributionSet",
+                        "desc", 2, "name==*", distributionSet, "50", "80"));
+    }
+
+    @Test
+    @Description("Verifies that an exception is thrown when trying to create a rollout with an incomplete distribution set.")
+    public void createRolloutWithIncompleteDistributionSet() {
+        final DistributionSet distributionSet = testdataFactory.createIncompleteDistributionSet();
+
+        assertThatExceptionOfType(IncompleteDistributionSetException.class)
+                .as("Incomplete distributionSet should throw an exception")
+                .isThrownBy(() -> testdataFactory.createRolloutByVariables("createRolloutWithIncompleteDistributionSet",
+                        "desc", 2, "name==*", distributionSet, "50", "80"));
+    }
+
+    @Test
+    @Description("Verifies that an exception is thrown when trying to update a rollout with an invalidated distribution set.")
+    public void updateRolloutWithInvalidDistributionSet() {
+        final DistributionSet distributionSet = testdataFactory.createDistributionSet();
+        testdataFactory.createTarget();
+        final Rollout rollout = testdataFactory.createRolloutByVariables("updateRolloutWithInvalidDistributionSet",
+                "desc", 2, "name==*", distributionSet, "50", "80");
+        final DistributionSet invalidDistributionSet = testdataFactory.createAndInvalidateDistributionSet();
+
+        assertThatExceptionOfType(InvalidDistributionSetException.class)
+                .as("Invalid distributionSet should throw an exception").isThrownBy(() -> rolloutManagement
+                        .update(entityFactory.rollout().update(rollout.getId()).set(invalidDistributionSet.getId())));
+    }
+
+    @Test
+    @Description("Verifies that an exception is thrown when trying to update a rollout with an incomplete distribution set.")
+    public void updateRolloutWithIncompleteDistributionSet() {
+        final DistributionSet distributionSet = testdataFactory.createDistributionSet();
+        testdataFactory.createTarget();
+        final Rollout rollout = testdataFactory.createRolloutByVariables("updateRolloutWithIncompleteDistributionSet",
+                "desc", 2, "name==*", distributionSet, "50", "80");
+        final DistributionSet incompleteDistributionSet = testdataFactory.createIncompleteDistributionSet();
+
+        assertThatExceptionOfType(IncompleteDistributionSetException.class)
+                .as("Incomplete distributionSet should throw an exception").isThrownBy(() -> rolloutManagement.update(
+                        entityFactory.rollout().update(rollout.getId()).set(incompleteDistributionSet.getId())));
+    }
+
```

Comment:
missing tests
- deleteRolloutWithInvalidDistributionSet
- deleteRolloutWithIncompleteDistributionSet

Reply:
Are these tests really needed? I think it should be no problem to delete a rollout with an invalidated or incomplete distribution set